server/backup_runner.go

```diff
@@ -296,6 +296,11 @@ func (r *BackupRunner) startAutoBackups(ctx context.Context, runningTasks map[in
 
 	for _, backupSetting := range backupSettingList {
 		mu.Lock()
+		// Only allow one running backup task.
+		if len(runningTasks) > 0 {
```

Comment:
I don't think this to be the intention. We are restricting one running backup per database instead of one running backup globally.

Reply:
Otherwise, it really defects the purpose having a for-loop here and running tasks.

Reply:
> We are restricting one running backup per database

IMO, it's what the current implementation does.

`CREATE UNIQUE INDEX idx_backup_setting_unique_database_id ON backup_setting(database_id);` ensures that backup setting is 1 to 1 mapped to database. We skip the backup schedule if there's one in the map of the running task.